src/onboarding/containers/CloudLoginPage.tsx

```diff
@@ -59,7 +59,11 @@ export const CloudLoginPage: FC = () => {
         fetch('/api/env/quartz-login-url')
           .then(async response => {
             const quartzUrl = await response.text()
-            const pathname = window.location.pathname ?? ''
+            const redirectTo = new URLSearchParams(window.location.search).get(
+              'redirectTo'
+            )
+            const pathname =
+              new URL(redirectTo).pathname ?? window.location.pathname ?? ''
```

Comment:
If there is `redirectTo` present, use it's pathname, else use the current window.location's pathname.